rumqttd/src/lib.rs

```diff
@@ -192,3 +195,14 @@ pub enum MetricType {
 pub struct MetricSettings {
     push_interval: u64,
 }
+
+#[derive(Debug, Serialize, Deserialize, Clone, Default)]
+pub enum SubProtocol {
```

Comment:
i think we are using this in context of Supported protocols only, so the name `SubProtocol` doesnt make sense. Maybe we can name it `SupportedProtocol` itself?

Reply:
SupportedProtocol makes sense. Will change that later today 👍🏼